adapter/integration-test/src/adapter-datasource.test.js

```diff
@@ -1,49 +1,75 @@
 const request = require('supertest')
 
 const {
-  ADAPTER_URL
+  ADAPTER_URL,
+  AMQP_URL
 } = require('./env')
 const { waitForServicesToBeReady } = require('./waitForServices')
+const {
+  connectAmqp,
+  consumeAmqpMsg
+} = require('./testHelper')
+
+const AMQP_EXCHANGE = 'ods_global'
+const AMQP_QUEUE = 'adapter-it-ds'
+const CONFIG_TOPIC = 'datasource.config.*'
+const CONFIG_CREATED_TOPIC = 'datasource.config.created'
+const CONFIG_DELETED_TOPIC = 'datasource.config.deleted'
+const CONFIG_UPDATED_TOPIC = 'datasource.config.updated'
 
 const TIMEOUT = 10000
+const publishedEvents = new Map() // routing key -> received msgs
+let amqpConnection
 
 describe('Datasource Configuration', () => {
   beforeAll(async () => {
     await waitForServicesToBeReady()
+
+    amqpConnection = await connectAmqp(AMQP_URL)
+
+    await consumeAmqpMsg(amqpConnection, AMQP_EXCHANGE, CONFIG_TOPIC, AMQP_QUEUE, publishedEvents)
   }, 60000)
 
+  afterAll(async () => {
+    await request(ADAPTER_URL)
+      .delete('/configs')
+      .send()
+    if (amqpConnection) {
+      await amqpConnection.close()
+    }
+  }, TIMEOUT)
+
   test('Should respond with stored datasource configs', async () => {
     const response = await request(ADAPTER_URL).get('/datasources')
     expect(response.status).toEqual(200)
     expect(response.type).toEqual('application/json')
 
-    expect(response.body).toEqual([])
+    expect(response.body).toBeInstanceOf(Array)
   }, TIMEOUT)
 
   test('Should create datasources [POST /datasources]', async () => {
     const response = await request(ADAPTER_URL)
       .post('/datasources')
       .send(datasourceConfig)
+    const datasourceId = response.body.id
 
     expect(response.status).toEqual(201)
     expect(response.header.location).toContain(response.body.id)
     expect(response.body.adapter).toEqual(datasourceConfig.adapter)
     expect(response.body.trigger).toEqual(datasourceConfig.trigger)
     expect(response.body.id).toBeDefined()
-    expect(response.body.id).not.toEqual(datasourceConfig.id) // id not under control of client
+    expect(datasourceId).not.toEqual(datasourceConfig.id) // id not under control of client
```

Comment:
If the `id` is given by the server `datasourceConfig` should not have a reasonable`id`.
Possible values could be `undefined`, `0`, or `-1`.
So you could test if a positive number was given as `id`.

In the current case a `-1234` value would be a pass, but should be a fail.

Reply:
You mean we should return `400 Bad Request: Field id should not be set` instead of just ignoring it?

Reply:
I will change the adapter service so it does not accept post requests with specified id fields